internal/util/iterator/iterator.go

```diff
@@ -44,5 +44,5 @@ type Interface[K, V any] interface {
 	//
 	// Close must be concurrency-safe and may be called multiple times.
```

Comment:
We should probably fix the description since we changed from function to interface.